✨ Add option to trim commit message to prevent hitting env var size limit

JeromeFitz opened the pull request:
This limits to  the first line of the commit (subject).

---

**Use Cases**:

Depending on the size of the `commit.commitMessage`
 it may set off the Environment Variable size limit on Vercel:

```sh
Error! The total size of all Environment Variables (9.1KB) exceeds 4KB
```

- A `Preview` deployment from a PR may take the latest commit `sha`
  - Chances are with conventional (gitmoji) commits, the overall `commit.message` may not introduce this problem
- A `Production` deployment from `main` may be a squash/merge commit with detailed PR information (images, logs, etc.)
  - Looking at you `dependabot` 🤖, and people like `@JeromeFitz` who write long PR messages 😆

JeromeFitz committed:
♻️ trim `commitMessage` to first line (subject)

Depending on the size of the `commit.commitMessage`
 it may set off the Environment Variable size limit on Vercel:

```sh
Error! The total size of all Environment Variables (9.1KB) exceeds 4KB
```

This limits to `50` characters which should encompass just the
 subject line of the commit message.

JeromeFitz commented:
Thanks for this wonderful :octocat: GitHub Action @BetaHuhn  💯 

This has been very helpful to have CI/CD run  _before_ moving to 🔺Vercel.

For Production deployments, I kept running into random errors on the deployment due to Environment Variable sizing, however, had under 10 on the particular Organization/Project. This would happen with some commits and not others.

Here is an example of it working:
- PR: https://github.com/JeromeFitz/websites/pull/887
  - Large PR Commit Message which would ballon the `githubCommitMessage` 🎈 
- Actions: https://github.com/JeromeFitz/websites/runs/5116392439?check_suite_focus=true

BetaHuhn commented on src/github.js:
```diff
@@ -105,7 +105,7 @@ const init = () => {
 			authorName: data.commit.author.name,
 			authorLogin: data.author.login,
-			commitMessage: data.commit.message
+			commitMessage: data.commit.message.substring(0, 50)
```
This isn't really needed since you trim it later anyway.

BetaHuhn commented on src/vercel.js:
```diff
@@ -36,7 +36,7 @@ const init = () => {
 				`githubCommitAuthorName=${ commit.authorName }`,
 				`githubCommitAuthorLogin=${ commit.authorLogin }`,
-				`githubCommitMessage=${ commit.commitMessage }`,
+				`githubCommitMessage=${ commit.commitMessage.substring(0, 50) }`,
```
Why did you choose to limit the commit message to exactly 50 characters? The 4kb limit should allow for a fair amount of text. As you mentioned, the first line of a commit message is often used as a subject, but it is not fixed in length (50 is just a convention).

Why not select the message up to the first line break? That way we can limit the message to the "subject line" however long it might be. Something like `commitMessage.split(/\r?\n/)[0]` should work.

JeromeFitz replied:
Awesome suggestion! The `50` came from a conventional commit pithy subject, but that should not be imposed here. Will update. 👍🏻

BetaHuhn reviewed (changes requested):
Hi @JeromeFitz, thanks for taking the time to create this PR and for the nice words!

I have never encountered the environment variables limit on Vercel, but I usually also don't have long commit messages :D

I left some comments as a review.

JeromeFitz commented:
Thanks for the review @BetaHuhn , will make the requested updates. 💯 

Would this be better served as a configuration setting? 

```yml
  TRIM_COMMIT_MESSAGE:
    description: |
      When passing meta data to Vercel deployment, trim the commit message to subject only (default: false).
    required: false
```

Not sure you would think this as additional bloat. If you have not run into this, or others, then this could be more "opt-in" until more feedback is gathered to make as a default setting.

BetaHuhn commented:
> Would this be better served as a configuration setting?

Yes, good idea!l Let's make this an optional action input. 

Also, please remove the new lines you added at the end of files. Otherwise the linter will fail.

JeromeFitz committed:
✨️ `TRIM_COMMIT_MESSAGE` setting (default: false)

Only `TRIM_COMMIT_MESSAGE` when sending to Vercel when this
 setting/option is `true`.

♻️  dist update

JeromeFitz commented:
Sweet. Updates are made, I'll be running a new deployment test soon. 💯 

Lint:
```sh
▲ deploy-to-vercel-action [ci/vercel-deploy-commit-limits] npm run lint

> deploy-to-vercel-action@1.5.11 lint /Users/jerome/Code/#__hacking/deploy-to-vercel-action
> eslint ./src

▲ ...
```

EDIT: The updated title is 64 characters, so on squash/merge this would have been truncated w/ the initial update prior to review 😆 

**Testing:**

Without `TRIM_COMMIT_MESSAGE` being passed (was passing to the wrong branch for the action, heh):

<img width="404" alt="image" src="https://user-images.githubusercontent.com/3099369/153716115-63d3ac0b-4af6-4ba9-885e-514cdf4a0816.png">



With `TRIM_COMMIT_MESSAGE` being passed:

<img width="405" alt="image" src="https://user-images.githubusercontent.com/3099369/153717010-50497003-3d64-4088-af1c-3769e58cff21.png">



Good to go for review!

JeromeFitz committed:
📝️ README update for `TRIM_COMMIT_MESSAGE`

BetaHuhn commented:
Awesome, thanks!

> The updated title is 64 characters, so on squash/merge this would have been truncated w/ the initial update prior to review 

😉
